MetadataExtractor/IO/BufferReader.cs

```diff
@@ -135,4 +138,250 @@ public string GetString(int bytesRequested, Encoding encoding)
+    }
+
+    public ushort GetUInt16(int index)
+    {
+        ValidateIndex(index, 2);
+
+        var bytes = _bytes.Slice(index, 2);
+
+        return _isBigEndian
+            ? BinaryPrimitives.ReadUInt16BigEndian(bytes)
+            : BinaryPrimitives.ReadUInt16LittleEndian(bytes);
+    }
+
+    public int GetInt24(int index)
+    {
+        Span<byte> bytes = stackalloc byte[3];
+
+        GetBytes(index, bytes);
+
+        if (_isBigEndian)
+        {
+            return
+                bytes[0] << 16 |
+                bytes[1] << 8 |
+                bytes[2];
+        }
+        else
+        {
+            return
+                bytes[2] << 16 |
+                bytes[1] << 8 |
+                bytes[0];
+        }
+    }
+
+    public int GetInt32(int index)
+    {
+        ValidateIndex(index, 4);
+
+        var bytes = _bytes.Slice(index, 4);
+
+        return _isBigEndian
+            ? BinaryPrimitives.ReadInt32BigEndian(bytes)
+            : BinaryPrimitives.ReadInt32LittleEndian(bytes);
+    }
+
+    public uint GetUInt32(int index)
+    {
+        ValidateIndex(index, 4);
+
+        var bytes = _bytes.Slice(index, 4);
+
+        return _isBigEndian
+            ? BinaryPrimitives.ReadUInt32BigEndian(bytes)
+            : BinaryPrimitives.ReadUInt32LittleEndian(bytes);
+    }
+
+    public float GetS15Fixed16(int index)
+    {
+        ValidateIndex(index, 4);
+
+        ReadOnlySpan<byte> bytes = _bytes.Slice(index, 4);
+
+        if (_isBigEndian)
+        {
+            float res = bytes[0] << 8 | bytes[1];
+            var d = bytes[2] << 8 | bytes[3];
+            return (float)(res + d / 65536.0);
+        }
+        else
+        {
+            // this particular branch is untested
+            var d = bytes[1] << 8 | bytes[0];
+            float res = bytes[3] << 8 | bytes[2];
+            return (float)(res + d / 65536.0);
+        }
+    }
+
+    public long GetInt64(int index)
+    {
+        ValidateIndex(index, 8);
+
+        var bytes = _bytes.Slice(index, 8);
+
+        return _isBigEndian
+            ? BinaryPrimitives.ReadInt64BigEndian(bytes)
+            : BinaryPrimitives.ReadInt64LittleEndian(bytes);
+    }
+
+    /// <exception cref="IOException"/>
+    public float GetFloat32(int index)
+    {
+#if NET462 || NETSTANDARD1_3
+        return BitConverter.ToSingle(BitConverter.GetBytes(GetInt32(index)), 0);
+#else
+        Span<byte> bytes = stackalloc byte[4];
+
+        GetBytes(index, bytes);
+
+        if (_isBigEndian)
+        {
+            bytes.Reverse();
+        }
+
+        return BitConverter.ToSingle(bytes);
+#endif
+    }
+
+    public double GetDouble64(int index)
+    {
+#if NET462 || NETSTANDARD1_3
+        return BitConverter.Int64BitsToDouble(GetInt64(index));
+#else
+        Span<byte> bytes = stackalloc byte[8];
+
+        GetBytes(index, bytes);
+
+        if (_isBigEndian)
+        {
+            bytes.Reverse();
+        }
+
+        return BitConverter.ToDouble(bytes);
+#endif
```

Comment:
.NET 8 (which is now a target) has `BinaryPrimitives.ReadDoubleBigEndian`/`LittleEndian` (also the `Single` variety), which could be used here (and in the non-indexed overload) too.

Reply:
done.